### PR TITLE
Key Search functionality

### PR DESCRIPTION
**Pull request by harshit777**

### CODE REVIEW

File: keys.go [hunk 1]
```diff
@@ -81,6 +81,7 @@ type Key struct {
 	PurgeAllowedFrom    *time.Time  `json:"purgeAllowedFrom,omitempty"`
 	PurgeScheduledOn    *time.Time  `json:"purgeScheduledOn,omitempty"`
 	DualAuthDelete      *DualAuth   `json:"dualAuthDelete,omitempty"`
+	Search              string      `json:"sort,omitempty"`
```
Comment (dinesh-venkatraman-g) [inline keys.go:84]: why it is `sort` here?

File: README.md [hunk 2]
```diff
@@ -213,4 +213,23 @@ if err != nil {
     fmt.Println(err)
 }
 fmt.Println(keys)
+```
+
+### Using Search functionality in list Keys API
```
Comment (dinesh-venkatraman-g) [inline README.md:218]: Can you add a comment with the link to API docs that explain how the search query is getting formed.

**Review by dinesh-venkatraman-g — CHANGES_REQUESTED** [review]
Can you add test reports? Evidences for new feature and evidences that all the existing testcase are passing.

File: kp_test.go [hunk 3]
```diff
@@ -4145,3 +4145,120 @@ func TestSyncAssociatedResourcesError(t *testing.T) {
...
+
+	searchStr := "aliasNew"
+	srcStr2, _ := GetKeySearchQuery(&searchStr, AddAliasScope())
+	listKeysOptions := &ListKeysOptions{
+		Search: srcStr2,
+	}
+
+	keys, err := c.ListKeys(context.Background(), listKeysOptions)
+	assert.NoError(t, err)
+	assert.NotNil(t, keys)
```
Comment (mrodden) [inline kp_test.go:4207]: Need to assert also that the call was actually made

```suggestion
	assert.NotNil(t, keys)
	assert.True(t, gock.IsDone(), "Expected HTTP requests not called")
```

File: kp_test.go [hunk 4]
```diff
@@ -4145,3 +4145,120 @@ func TestSyncAssociatedResourcesError(t *testing.T) {
...
+	srcStr2, _ = GetKeySearchQuery(&searchStr, AddEscape())
+	fmt.Printf("%s\n", *srcStr2)
+
+	listKeysOptions = &ListKeysOptions{
+		Search: srcStr2,
+	}
+
+	keys, err = c.ListKeys(context.Background(), listKeysOptions)
+	assert.NoError(t, err)
+	assert.NotNil(t, keys)
```
Comment (mrodden) [inline kp_test.go:4262]: Here as well

```suggestion
	assert.NotNil(t, keys)
	assert.True(t, gock.IsDone(), "Expected HTTP requests not called")
```

**Review by mrodden — CHANGES_REQUESTED** [review]
This looks like it would sit in the `listkeys.go` file well with the new sorting functionality from #80

File: keys.go (hunk 1 above)
Comment (harshit777) [reply keys.go:84]: Seems like typo issue here, will fix this

**Comment by harshit777** [comment]
Screenshot 2022-07-31 at 12 13 11 PM


Screenshot 2022-07-31 at 12 15 05 PM

Screenshot 2022-07-31 at 12 16 31 PM

**Review by dinesh-venkatraman-g — APPROVED** [review]
1. Keep the key search related changes after sort related changes in listkeys.go file so that merge will be easier
2. Add test evidences with this change

**Comment by harshit777** [comment]
Result of >>    **go test -v -tags=integration ./...** 


Screenshot 2022-08-04 at 1 53 22 PM

**Review by dinesh-venkatraman-g — APPROVED** [review]
lgtm

File: keys.go [hunk 5]
```diff
@@ -80,6 +80,7 @@ type Key struct {
 	PurgeAllowedFrom    *time.Time  `json:"purgeAllowedFrom,omitempty"`
 	PurgeScheduledOn    *time.Time  `json:"purgeScheduledOn,omitempty"`
 	DualAuthDelete      *DualAuth   `json:"dualAuthDelete,omitempty"`
+	Search              string      `json:"search,omitempty"`
```
Comment (mrodden) [inline keys.go:83]: I don't understand why we are adding and extra field to the Key data that comes from the service. I don't see it in any of the documentation either.

**Review by mrodden — CHANGES_REQUESTED** [review]

**Comment by dinesh-venkatraman-g** [comment]
Already covered in another PR

**Pull request closed without merging**